libraries/ESP8266WiFi/src/include/UdpContext.h

```diff
@@ -84,11 +84,9 @@ class UdpContext
 
     void unref()
     {
-        if(this != 0) {
```

Comment:
I've been reading up on this, and it turns out that this check is useless, because (newer) compilers are free to optimize it away due to the nonnull attribute added to the this ptr, which also happens fo be the cause of a compile time warning/error.